src/main/java/com/faforever/client/replay/ReplayService.java

```diff
@@ -201,8 +200,21 @@ private void onLocalReplaysWatchEvent(WatchKey key) {
       }
     }
 
-    localReplays.addAll(newReplays);
-    publisher.publishEvent(new LocalReplaysChangedEvent(this, newReplays, deletedReplays));
+    CompletableFuture[] replayFuturesArray = newReplaysFutures.toArray(new CompletableFuture[newReplaysFutures.size()]);
+    CompletableFuture<List<Replay>> newReplaysFuture = CompletableFuture.allOf(replayFuturesArray)
+        .thenApply(ignoredVoid ->
+            newReplaysFutures.stream()
+                .map(future -> future.join())
```

Comment:
CompletableFuture
        .allOf

Reply:
What?

Reply:
To join CompletedFutures use CompletableFuture.allOf()

Reply:
Which is exactly what I'm doing. Hence my confusion. :)

Reply:
Should read the code closer sorry lemme check again